src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallerFactory.cs

```diff
@@ -38,7 +39,7 @@ internal class WorkloadInstallerFactory
                 return new NetSdkMsiInstaller();
             }
 
-            if (!CanWriteToDotnetRoot(dotnetDir))
+            if (elevationRequired && !CanWriteToDotnetRoot(dotnetDir))
```

Comment:
I think there are too many flags for the logic. Any reason we cannot catch the exception and point the error only when we failed to write?

Reply:
The `CanWriteToDotnetRoot` method does catch the error and it returns false only when we failed to write. The issue here is that there are some options (namely the print download like option) that don't need to write to the dotnet root, so they don't need to be elevated and this is erroring unnecessarily.

Reply:
@wli3 do you have an alternative suggestion here?

Reply:
Just don't do any explicit check, rely on catch and rethrow better error message?

Reply:
I originally added this explicit check because we wanted to error right away if we had inadequate permissions, if we wait until we try to access the dotnet root in the actual installation then we've already done a lot of work (updating ad manifests, downloading/ extracting packages). Also, if we want to catch and rethrow with a better error message whenever we run into a permission issue, we'll have to duplicate that across quite a few places, since there are lots of places we could run into that issue.